plugins/content_distribution/providers/pushToNews/lib/PushToNewsDistributionEngine.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * @package plugins.pushToNewsDistribution
+ * @subpackage lib
+ */
+class PushToNewsDistributionEngine extends DistributionEngine implements 
+	IDistributionEngineSubmit,
+	IDistributionEngineUpdate,
+	IDistributionEngineDelete
+{
+	const CERT_KEY_FILE_NAME = "cert.pem";
+	const TEMP_DIRECTORY = 'push_to_news_distribution';
+
+	/**
+	 * @see IDistributionEngineSubmit::submit()
+	 */
+	public function submit(KalturaDistributionSubmitJobData $data)
+	{
+		return $this->submitOrUpdate($data);
+	}
+	
+	/**
+	 * @see IDistributionEngineUpdate::update()
+	 */
+	public function update(KalturaDistributionUpdateJobData $data)
+	{
+		return $this->submitOrUpdate($data);
+	}
+	
+	public function submitOrUpdate(KalturaDistributionJobData $data)
+	{
+		$this->validateJobObjects($data);
+
+		$distributionProfile = $data->distributionProfile;
+		$host = $distributionProfile->host;
+		$port = $distributionProfile->port;
+		$password = $distributionProfile->password;
+		$basePath = $distributionProfile->basePath;
+		
+		$url = $host;
```

Comment:
Need to move this to a helper function that will receive `$distributionProfile` as a parameter and return the URL.
Same goes for https://github.com/kaltura/server/pull/4476/files#diff-89ba83b8ce3c12e749ec8357f423036dR74